pages/talks.js

```diff
@@ -1,52 +1,49 @@
 import React from 'react';
 
-import fetch from '../lib/fetch';
+import useDataApi from '../lib/useDataApi';
 import Layout from '../components/Layout';
 import Talk from '../components/Talk';
 import { CardWithLoading } from '../components/Card';
 import Grid from '../components/Grid';
 
-const Talks = () => {
-    const [calendar, setCalendar] = React.useState(null);
-    const [noDate, setNoDate] = React.useState(null);
-
-    React.useEffect(() => {
-        const fetchTalks = async () => {
-            const { calendar, notPlanned, error } = await fetch('/calendar');
-            if (error) {
-                return console.error(error);
-            }
-
-            setCalendar(calendar);
-            setNoDate(notPlanned);
-        };
-
-        fetchTalks();
-    }, []);
-
-    const loadingCalendar = [...Array(6).keys()]
+const LoadingCalendar = () =>
+    [...Array(6).keys()]
         .map((_, index) => <CardWithLoading key={`CardWithLoading${index}`} />)
         .concat(
             [...Array(6).keys()].map((_, index) => (
                 <CardWithLoading key={`CardWithLoadingConcat${index}`} />
             )),
         );
 
+const Talks = () => {
+    const {
+        data: { calendar, notPlanned },
+        isLoading,
+        isError,
+    } = useDataApi('/api/calendar', {
+        calendar: [],
+        notPlanned: [],
+    });
+
+    const isLoadingElement = isLoading || isError;
```

Comment:
Same as in `index.js`, the error state is hidden and not handled

Reply:
What did you sugest in case of error ? Empty card with error image (https://codepen.io/juliepark/pen/vjMOKQ) ? Wee can create an issue

Reply:
A simple error message would work as well

Reply:
Created #25